src/sortStudents.ts

```diff
@@ -1,16 +1,62 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder = 'asc',
+): Student[] {
+  const copyStudents = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return copyStudents.sort((a, b) => {
+    switch (sortBy) {
+      case SortType.Name:
+        return (order === 'asc')
+          ? a.name.localeCompare(b.name)
+          : b.name.localeCompare(a.name);
+
+      case SortType.Surname:
+        return (order === 'asc')
+          ? a.surname.localeCompare(b.surname)
+          : b.surname.localeCompare(a.surname);
+
+      case SortType.Age:
+        return (order === 'asc')
+          ? a.age - b.age
+          : b.age - a.age;
+
+      case SortType.Married:
+        return (order === 'asc')
+          ? +a.married - +b.married
+          : +b.married - +a.married;
```

Comment:
Use Number() constructor instead, it is more obvious
```suggestion
          ? Number(a.married) - Numberb.married)
          : Number(b.married - Number(a.married);
```